cli/azd/cmd/init.go

```diff
@@ -79,9 +75,37 @@ func (i *initAction) SetupFlags(
 		"Name or ID of an Azure subscription to use for the new environment",
 	)
 	local.StringVarP(&i.location, "location", "l", "", "Azure location for the new environment")
+
+	i.rootOptions = global
 }
 
-func (i *initAction) Run(ctx context.Context, cmd *cobra.Command, args []string, azdCtx *azdcontext.AzdContext) error {
+type initAction struct {
+	azdCtx  *azdcontext.AzdContext
+	console input.Console
+	cmdRun  exec.CommandRunner
+	azCli   azcli.AzCli
+	gitCli  git.GitCli
+	flags   initFlags
+}
+
+func newInitAction(
+	azdCtx *azdcontext.AzdContext,
+	cmdRun exec.CommandRunner,
+	console input.Console,
+	azCli azcli.AzCli,
+	gitCli git.GitCli,
+	flags initFlags) (*initAction, error) {
+	return &initAction{
+		azdCtx:  azdCtx,
+		console: console,
+		cmdRun:  cmdRun,
+		azCli:   azCli,
+		gitCli:  gitCli,
+		flags:   flags,
+	}, nil
+}
```

Comment:
I would like to see us split out the `Action` implementations from the `Cmd` into separate go packages.

1. Command structs design the cobra command layer, instantiate the Action via Wire and exec the action
2. Actions don't even know comands exist and just have constructor `NewXXX(...)` command with injected dependencies (via Wire).

Reply:
That is how it is currently designed. The only thing is that the `action` class lives in the command files. We can discuss if we want to move it out (I'm personally in favor), but it should be done separately to minimize code diffs.

Reply:
Great, lets create an issue to track this so we don't lose track.

Reply:
#900